### PR TITLE
Fix issue with inaccessible helpers from nested components

### PR DESCRIPTION
**Pull request by franco**
As reported in #321, global Rails helpers are not accessible within
nested components. This is because their view context is the parent
component, where global helpers are not available.

### CODE REVIEW

File: lib/view_component/base.rb [hunk 1]
```diff
@@ -108,7 +108,7 @@ def controller
 
     # Provides a proxy to access helper methods
     def helpers
-      @helpers ||= view_context
+      @helpers ||= view_context.controller.view_context
```
Comment (joelhawksley) [inline lib/view_component/base.rb:111]:
```suggestion
      @helpers ||= controller.view_context
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # master
 
+* Fix bug where global Rails helpers are inaccessible from nested components.
+
+    *Franco Sebregondi*
+
+
```
Comment (joelhawksley) [inline CHANGELOG.md:7]:
```suggestion
```

File: test/view_component/view_component_test.rb [hunk 3]
```diff
@@ -221,6 +221,12 @@ def test_renders_helper_method_through_proxy
     assert_text("Hello helper method")
   end
 
+  def test_renders_helper_method_within_nested_component
+    render_inline(ContainerComponent.new(nested_component: HelpersProxyComponent.new))
```
Comment (joelhawksley) [inline test/view_component/view_component_test.rb:225]:
```suggestion
    render_inline(ContainerComponent.new)
```

File: test/app/components/container_component.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class ContainerComponent < ViewComponent::Base
+  def initialize(nested_component:)
+    @nested_component = nested_component
+  end
```
Comment (joelhawksley) [inline test/app/components/container_component.rb:6]:
```suggestion
```

File: test/app/components/container_component.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+  def call
+    render @nested_component
```
Comment (joelhawksley) [inline test/app/components/container_component.rb:9]:
```suggestion
    render HelpersProxyComponent.new
```

File: test/app/components/container_component.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class ContainerComponent < ViewComponent::Base
```
Comment (joelhawksley) [inline test/app/components/container_component.rb:3]:
```suggestion
class HelpersContainerComponent < ViewComponent::Base
```

**Comment by joelhawksley** [comment]
@franco regarding your comment [here](https://github.com/github/view_component/issues/321#issuecomment-643925454), I don't think this is a breaking change. 

This method was always implied to function the same as calling `helpers.` in a controller, so I think this really just a wonderful bug fix ❤️. Perhaps you could add a note to the changelog entry describing the previous, incorrect behavior?

File: lib/view_component/base.rb [hunk 7]
```diff
@@ -108,7 +108,7 @@ def controller
 
     # Provides a proxy to access helper methods
```
Comment (joelhawksley) [inline lib/view_component/base.rb:109]:
```suggestion
    # Provides a proxy to access helper methods from the context of the current controller
```

**Comment by franco** [comment]
@joelhawksley thanks a lot for the suggestions. Much better 👍. I amended the changelog entry.

File: test/app/components/container_component.rb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class HelpersContainerComponent < ViewComponent::Base
+
```
Comment (joelhawksley) [inline test/app/components/container_component.rb:4]:
```suggestion
```

**Comment by joelhawksley** [comment]
@franco thank you for your contribution! I look forward to your next PR! ❤️

**Pull request merged**